bottlerocket/agents/src/bin/ecs-workload-agent/main.rs

```diff
@@ -0,0 +1,545 @@
+
+        info!("Waiting for registered container instances...");
+
+        tokio::time::timeout(
+            Duration::from_secs(30),
+            wait_for_cluster_ready(&ecs_client, &self.config.cluster_name),
+        )
+        .await
+        .context(error::InstanceTimeoutSnafu)??;
+
+        // First loop through and make sure we have task definitions for each of our tests.
+        // We don't want to starts tasks if one of the defined tets is not able to be run.
+        let mut task_def_arns = Vec::new();
+        for test_def in &self.config.tests {
+            let task_def_arn = create_or_find_task_definition(&ecs_client, test_def).await?;
+
+            task_def_arns.push(task_def_arn);
+        }
+
+        let mut results = TestResults {
+            outcome: Outcome::InProgress,
+            other_info: Some("Starting test".to_string()),
+            ..Default::default()
+        };
+        info_client
+            .send_test_update(results.clone())
+            .await
+            .err()
+            .iter()
+            .for_each(|e| log::error!("Unable to send test update: {}", e));
+
+        // Now that we have the full set of task definitions to run, loop through and kick them all off.
+        let mut task_arns = Vec::new();
+        for task_def_arn in task_def_arns {
+            info!("Running task '{}'", task_def_arn);
+            let run_task_output = ecs_client
+                .run_task()
+                .cluster(&self.config.cluster_name)
+                .task_definition(task_def_arn)
+                .launch_type(LaunchType::Ec2)
+                .send()
+                .await
+                .context(error::TaskRunCreationSnafu)?;
+            let run_task_arns: Vec<String> = run_task_output
+                .tasks()
+                .map(|tasks| {
+                    tasks
+                        .iter()
+                        .filter_map(|task| task.task_arn().map(|arn| arn.to_string()))
+                        .collect()
+                })
+                .unwrap();
+            for task_arn in &run_task_arns {
+                task_arns.push(task_arn.clone());
+            }
+        }
+
+        info!("Waiting for tasks to complete...");
+        results.other_info = Some("Checking status".to_string());
+        info_client
+            .send_test_update(results)
+            .await
+            .err()
+            .iter()
+            .for_each(|e| log::error!("Unable to send test update: {}", e));
+
+        match tokio::time::timeout(
+            Duration::from_secs(30),
+            wait_for_test_completion(
+                &ecs_client,
+                &self.config.cluster_name,
+                &task_arns,
+                info_client,
+            ),
+        )
+        .await
+        {
+            Ok(results) => results,
+            Err(_) => test_results(&ecs_client, &self.config.cluster_name, &task_arns).await,
+        }
```

Comment:
Does this mean the task only has 30s to run to completion?

Reply:
Hmm, yeah, that's way too short for these. Good catch.